app/javascript/controllers/leaflet_viewer_controller.js

```diff
@@ -76,6 +76,7 @@ export default class LeafletViewerController extends Controller {
       await this.addPreviewOverlay();
       this.addInspection();
     } else if (this.drawInitialBoundsValue && this.bounds) {
+      if (this.mapGeomValue) this.map.addLayer(L.geoJSON(this.mapGeomValue))
```

Comment:
is this different from line 50 because it creates an actual `geoJSON` layer, instead of just returning the bounds of that layer? maybe we should just change line 50 to do that directly, instead of doing it here, since it would work just as well for non-complex items. that avoids the scenario where we have two different layers, one for the bounds (non-complex) and one for the entire geometry (complex).

Reply:
actually, it might be cleaner to just move all this logic to `addBoundsOverlay()`. I think we should only draw one overlay.

Reply:
happy to talk/pair on this to explain what I'm thinking

Reply:
Sorry! Are you looking to pair with @dnoneill or me? This PR is a bit of a mashup now.

Reply:
@thatbudakguy // Moved the geom render into addBoundsOverlay. Simplifies that if/else block into calling one function to render added boundary layers. LMKWYT.